test/gtest/ucp/test_ucp_rma.cc

```diff
@@ -347,23 +331,83 @@ UCS_TEST_P(test_ucp_rma, get_blocking_zcopy, "ZCOPY_THRESH=0") {
 
 UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(test_ucp_rma)
 
-UCS_TEST_P(test_ucp_rma_reg_nb, put_blocking) {
-    test_message_sizes(static_cast<send_func_t>(&test_ucp_rma::put_b));
+
+class test_ucp_rma_reg : public test_ucp_rma {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant>& variants) {
+        add_variant_with_value(variants, UCP_FEATURE_RMA, 0, "");
+        add_variant_with_value(variants, UCP_FEATURE_RMA, NON_BLOCK, "nb");
+        add_variant_with_value(variants, UCP_FEATURE_RMA, NON_BLOCK | NO_DEVX, "no_devx");
+        add_variant_with_value(variants, UCP_FEATURE_RMA, NO_RCACHE, "no_rcache");
+    }
+
+    virtual void init() {
+        if (get_variant_value() & NON_BLOCK) {
+            modify_config("REG_NONBLOCK_MEM_TYPES", "host");
+        }
+        if (get_variant_value() & NO_DEVX) {
+            modify_config("IB_MLX5_DEVX_OBJECTS", "", SETENV_IF_NOT_EXIST);
+        }
+        if (get_variant_value() & NO_RCACHE) {
+            modify_config("RCACHE_ENABLE", "n");
+        }
+        test_ucp_rma::init();
+    }
+
+protected:
+    void test_reg(send_func_t send_func, bool host_only) {
```

Comment:
As I can see, `host_only` arg is always set to `host_only()` call. Can we check `host_only()` within the function and remove the argument?